src/main/scala/rocket/PTW.scala

```diff
@@ -199,66 +201,100 @@ class PTW(n: Int)(implicit edge: TLEdgeOut, p: Parameters) extends CoreModule()(
     "PTE Cache Hit/Miss Performance Monitor Events are lower priority than L2TLB Hit event")
 
   val l2_refill = RegNext(false.B)
+  val l2_flush_entry = RegNext(false.B)
   l2_refill_wire := l2_refill
+  l2_flush_entry_wire := l2_flush_entry
   io.dpath.perf.l2miss := false
   io.dpath.perf.l2hit := false
   val (l2_hit, l2_error, l2_pte, l2_tlb_ram) = if (coreParams.nL2TLBEntries == 0) (false.B, false.B, Wire(new PTE), None) else {
-    val code = new ParityCode
+    val nL2TLBSets = 32
```

Comment:
rather than hard-coding `nL2TLBSets = 32`, I will add a parameter `coreParams.nL2TLBWays` and calculate `val nL2TLBSets = coreParams.nL2TLBEntries/coreParams.nL2TLBWays`